infra/keyvault/application-keyvault.bicep

```diff
@@ -0,0 +1,52 @@
+@description('Required. The parameter object for the virtual network. The object must contain the name,skuName,resourceGroup and subnetPrivateEndpoints values.')
+param vnet object
+
+@description('Required. The parameter object for keyvault. The object must contain the name, enableSoftDelete, enablePurgeProtection and softDeleteRetentionInDays values.')
+param keyVault object
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. Environment name.')
+param environment string
+
+@description('Optional. Resource tags.')
+param tags object = {
+  Description: 'CDO Platform KeyVault Store'
+}
+
+@description('Required. Date in the format yyyy-MM-dd.')
+param createdDate string = utcNow('yyyy-MM-dd')
+
+var customTags = {
+  Location: location
+  CreatedDate: createdDate
+  Environment: environment
+}
+
+var defaultTags = json(loadTextContent('../default-tags.json'))
+var combinedTags = union(defaultTags, tags, customTags)
+
+module vaults 'br/SharedDefraRegistry:key-vault.vaults:0.5.6' = {
+  name: '${uniqueString(deployment().name, location)}-keyvault'
```

Comment:
@kbabigumira @osagiestar - Is it possible to change the deployment name to the convention we've used for AKS and ServiceBus for consistency?

e.g.
@description('Required. Date in the format yyyyMMdd-HHmmss.')
param deploymentDate string = utcNow('yyyyMMdd-HHmmss')

name: 'app-keyvault-${deploymentDate}'

Reply:
updated to match the convention used for the other resources